src/pages/distroSettings/tabs/HostTab/getFormSchema.ts

```diff
@@ -0,0 +1,257 @@
+import { GetFormSchema } from "components/SpruceForm";
+import { CardFieldTemplate } from "components/SpruceForm/FieldTemplates";
+import {
+  Arch,
+  BootstrapMethod,
+  CommunicationMethod,
+  FeedbackRule,
+  HostAllocatorVersion,
+  OverallocatedRule,
+  Provider,
+  RoundingRule,
+} from "gql/generated/types";
+
+type FormSchemaParams = {
+  provider: Provider;
+};
+
+export const getFormSchema = ({
+  provider,
+}: FormSchemaParams): ReturnType<GetFormSchema> => {
+  const hasStaticProvider = provider === Provider.Static;
+  const hasDockerProvider = provider === Provider.Docker;
+
+  return {
+    fields: {},
+    schema: {
+      type: "object" as "object",
+      properties: {
+        setup: {
+          type: "object" as "object",
+          title: "Host Setup",
+          properties: {
+            bootstrapMethod: {
+              type: "string" as "string",
+              title: "Host Bootstrap Method",
+              oneOf: enumSelect(bootstrapMethodToCopy),
+            },
+            communicationMethod: {
+              type: "string" as "string",
+              title: "Host Communication Method",
+              oneOf: enumSelect(communicationMethodToCopy),
+            },
+            arch: {
+              type: "string" as "string",
+              title: "Agent Architecture",
+              oneOf: enumSelect(architectureToCopy),
+            },
+            workDir: {
+              type: "string" as "string",
+              title: "Working Directory",
+            },
+            setupScript: {
+              type: "string" as "string",
+              title: "Setup Script",
+            },
+            setupAsSudo: {
+              type: "boolean" as "boolean",
+              title: "Run script as sudo",
+            },
+          },
+        },
+        sshConfig: {
+          type: "object" as "object",
+          title: "SSH Configuration",
+          properties: {
+            user: {
+              type: "string" as "string",
+              title: "SSH User",
+            },
+            sshKey: {
+              type: "string" as "string",
+              title: "SSH Key",
+            },
```

Comment:
One thing I'm unsure about is that the Legacy UI and the designs have this value as a dropdown 🤔 But I'm not sure if that's a mistake, or if they actually get the keys from somewhere in the backend

<img width="202" alt="Screenshot 2023-09-01 at 4 21 29 PM" src="https://github.com/evergreen-ci/spruce/assets/47064971/8c13f4b8-fa0d-43da-8600-13c9b20233b3">

Reply:
Whoops thank you! I opened https://github.com/evergreen-ci/evergreen/pull/6993 to expose this field.